akka-http-marshallers-scala/akka-http-spray-json/src/main/scala/akka/http/scaladsl/marshallers/sprayjson/SprayJsonByteStringParserInput.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.http.scaladsl.marshallers.sprayjson
+
+import java.nio.{ ByteBuffer, CharBuffer }
+import java.nio.charset.{ Charset, StandardCharsets }
+
+import akka.util.ByteString
+import spray.json.ParserInput.DefaultParserInput
+import scala.annotation.tailrec
+
+/**
+ * ParserInput reading directly off a ByteString. (Based on the ByteArrayBasedParserInput)
+ * This avoids a separate decoding step but assumes that each byte represents exactly one character,
+ * which is encoded by ISO-8859-1!
+ * You can therefore use this ParserInput type only if you know that all input will be `ISO-8859-1`-encoded,
+ * or only contains 7-bit ASCII characters (which is a subset of ISO-8859-1)!
+ *
+ * Note that this ParserInput type will NOT work with general `UTF-8`-encoded input as this can contain
```

Comment:
Oh, good catch though I think only the comment is wrong?
See https://github.com/akka/akka/pull/20778/files/895df968a91c3ac1df314f277ab4bf2fcf850b7b#diff-8ac9e4f50d3ee161f7717b3f17729874R37